genie-core/src/main/java/com/netflix/genie/core/jobs/workflow/impl/JobTask.java

```diff
@@ -101,7 +114,11 @@ public void executeTask(
         // Wait for the above process started in background mode. Wait lets us get interrupted by kill signals.
         Utils.appendToWriter(writer, "wait $!");
 
+        // Append new line
+        Utils.appendToWriter(writer, " ");
+
         // capture exit code and write to genie.done file
+        Utils.appendToWriter(writer, "# Write the return code from the command in the done file.");
         Utils.appendToWriter(writer, JobConstants.GENIE_DONE_FILE_CONTENT_PREFIX + JobConstants.GENIE_DONE_FILE_NAME);
```

Comment:
Instead of calling appendToWriter so many times why not build up a List<String> and pass the whole list at the end to the method once and just iterate through and write it out. Puts less function calls on the stack. Not sure if the compiler would optimize that out or not but seems like it might clean the code up a bit. Though possible that calling list.add would be the same.... not sure just seems like there's a lot of calls to this util method that maybe there is a cleaner way.